### PR TITLE
Supports multiple lanes per segment.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #16 
Goes on top of #27 

## Summary
- Supports multiple lanes per segment
- Segment ID:
  - Segment ids are created from a integer counter starting at zero.
- LaneID:
  - Lane ids are exactly the same as the lanelet's ids in the osm file.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: src/maliput_osm/osm/osm_manager.h [hunk 1]
```diff
@@ -63,6 +65,36 @@ class OSMManager {
...
+  //
+  // @param lane_id The lane_id to find the adjacent lanes of.
+  // @param lanes The lanes to search in.
+  // @param left Whether to search for the left or right adjacent lanes.
+  // @param[out] segment The segment to be filled. The lanes vector is filled from rightest most lane to leftest most
+  // lane.
+  //
+  // @throws maliput::common::assertion_error When the adjacent lane id doesn't match any lane in the @p lanes.
+  static void AddLanesToSegment(const Lane::Id& lane_id, const std::unordered_map<Lane::Id, Lane>& lanes, bool left,
+                                Segment& segment);
```
Comment (agalbachicar) [inline src/maliput_osm/osm/osm_manager.h:84]: `Segment& segment` should be `Segment* segment` due to our styleguide.

File: src/maliput_osm/osm/osm_manager.h [hunk 2]
```diff
@@ -63,6 +65,36 @@ class OSMManager {
...
+  // Creates the complete Segment for a given @p lane .
+  // The graph is traversed using the lane.left_lane_id and lane.right_lane_id in order to complete the Segment.
+  // When a lane already belongs to a segment then no segment is created and std::nullopt is returned.
+  // @param lane The lane to create a Segment from.
+  // @param lanes The map of all lanes.
+  // @returns A populated Segment or std::nullopt when the lane already belongs to a segment.
+  std::optional<Segment> CreateSegmentForLane(const Lane& lane, const std::unordered_map<Lane::Id, Lane>& lanes) const;
+
+  // Id generator for the segments.
+  mutable utilities::IdGen<Segment::Id> segment_id_gen_;
```
Comment (agalbachicar) [inline src/maliput_osm/osm/osm_manager.h:95]: Why is this `mutable`? This class is not expected to be mutable after construction as the parsing occurs in the constructor. Am I right?

**Review by agalbachicar — CHANGES_REQUESTED** [review]
A few comments to go. PTAL

File: src/maliput_osm/osm/osm_manager.h (hunk 2 above)
Comment (francocipollone) [reply src/maliput_osm/osm/osm_manager.h]: Fully agree. 
I'd wanted to make `CreateSegmentForLane` method const and that's why I added the `mutable` keyword to that variable however given that this method is only; used by the constructor then this doesn't affect to making a constant `OSMManager` instance.

Fix at [92c5f87](https://github.com/maliput/maliput_osm/pull/28/commits/92c5f87e9abcb3ce68b10c698d32c9c60a977d00)

File: src/maliput_osm/osm/osm_manager.h (hunk 1 above)
Comment (francocipollone) [reply src/maliput_osm/osm/osm_manager.h]: Done.

File: src/maliput_osm/osm/osm_manager.cc [hunk 3]
```diff
@@ -37,25 +37,70 @@
...
+}
+
+void OSMManager::AddLanesToSegment(const Lane::Id& lane_id, const std::unordered_map<Lane::Id, Lane>& lanes, bool left,
+                                   Segment* segment) {
+  const auto lane = lanes.find(lane_id);
+  MALIPUT_THROW_UNLESS(lane != lanes.end());
+  // Add lane to segment.
+  segment->lanes.insert(left ? segment->lanes.end() : segment->lanes.begin(), lane->second);
+  // Check the following adjacent lane.
+  const auto next_adjacent_lane = left ? lane->second.left_lane_id : lane->second.right_lane_id;
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_osm/osm/osm_manager.cc]: Done!

File: src/maliput_osm/osm/osm_manager.cc [hunk 4]
```diff
@@ -37,25 +37,70 @@
 
 namespace maliput_osm {
 namespace osm {
+namespace {}  // namespace
```
[the remark this answers is not in the document]
Comment (francocipollone) [reply src/maliput_osm/osm/osm_manager.cc]: Removed. Done.

**Review by agalbachicar — APPROVED** [review]
LGTM

**Pull request merged**